### PR TITLE
Fix install destination

### PR DESCRIPTION
**Pull request by YutoUchimi**
When I installed `jsk_recognition` via `apt`(i.e. rosdep install) and ran `source /opt/ros/${ROS_DISTRO}/setup.bash`, some sample launch files could not find in `/opt/ros/${ROS_DISTRO}/share`.

This PR will fix this problem.

After this PR, install ...
- programs in `scripts/` into ${CATKIN_PACKAGE_BIN_DESTINATION}
- `sample` and `test` dir into ${CATKIN_PACKAGE_SHARE_DESTINATION}

cf.
http://wiki.ros.org/catkin/CMakeLists.txt
http://www.ros.org/reps/rep-0122.html

### CODE REVIEW

**Comment by YutoUchimi** [comment]
@knorth55 Could you please review this if you have time?

File: jsk_pcl_ros/CMakeLists.txt [hunk 1]
```diff
@@ -461,17 +461,20 @@ install(TARGETS jsk_pcl_ros jsk_pcl_ros_base jsk_pcl_ros_moveit
   ARCHIVE DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
   LIBRARY DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION})
 install(DIRECTORY scripts/
-        DESTINATION ${CATKIN_PACKAGE_BIN_DESTINATION}
-        USE_SOURCE_PERMISSIONS
+  DESTINATION ${CATKIN_PACKAGE_BIN_DESTINATION}
+  USE_SOURCE_PERMISSIONS
 )
 
 install(DIRECTORY
```
Comment (knorth55) [inline jsk_pcl_ros/CMakeLists.txt]: make lines one line as `install (DIRECTORY config euslisp launch plugins ...`

**Comment by knorth55** [comment]
please format directory name as `node_scripts/` -> `node_scripts`
these two styles are mixed now.

File: jsk_pcl_ros/CMakeLists.txt (hunk 1 above)
Comment (YutoUchimi) [reply jsk_pcl_ros/CMakeLists.txt]: Thanks! Updated.

**Comment by YutoUchimi** [comment]
In fact, I use `node_scripts/` style on purpose.

On debian distibutions, `${CATKIN_PACAKGE_BIN_DESTINATION}` and `${CATKIN_PACAKGE_LIB_DESTINATION}` are both set to `${CMAKE_INSTALL_PREFIX}/lib/${PROJECT_NAME}`.
(cf. http://docs.ros.org/jade/api/catkin/html/user_guide/variables.html)

Unlike `share` space, in almost all packages, executable programs are located directly in `lib` space.
e.g.
```
$ ls /opt/ros/indigo/lib/rosbag/
bag2png.py    fix_md5sums.py         fixbag.py        play        topic_renamer.py
bagsort.py    fix_moved_messages.py  fixbag_batch.py  record
fastrebag.py  fix_msg_defs.py        makerule.py      savemsg.py
```
```
$ ls /opt/ros/indigo/lib/roscpp_tutorials/
add_two_ints_client        custom_callback_processing  listener_threaded_spin        notify_connect
add_two_ints_server        listener                    listener_unreliable           parameters
add_two_ints_server_class  listener_async_spin         listener_with_tracked_object  talker
anonymous_listener         listener_class              listener_with_userdata        time_api_sleep
babbler                    listener_multiple           node_handle_namespaces        timers
```
```
$ ls /opt/ros/indigo/share/rosbag/
cmake  package.xml
```
```
$ ls /opt/ros/indigo/share/roscpp_tutorials/
cmake  launch  package.xml  srv
```

**Comment by knorth55** [comment]
ok, then can you add comment for that?

**Comment by YutoUchimi** [comment]
I see. Then I will follow below guideline (JSK original).
- If a package has `setup.py` at the top level of the package and modules under `src/${PROJECT_NAME}/`, then each python executable should be installed into `${CATKIN_PACKAGE_BIN_DESTINATION}`.
    This respects ordinal installation style.
- If not, directories including python executables such as `scripts` and `node_scripts` should be installed into `${CATKIN_PACKAGE_SHARE_DESTINATION}`.
    This is for ROS-beginners who want to find that executables easily with `roscd <your package>; cd scripts`.

**Comment by k-okada** [comment]
https://travis-ci.org/jsk-ros-pkg/jsk_recognition/jobs/448650345
```
jsk_pcl_ros/test_results/jsk_pcl_ros/rosunit-test_octree_voxel_grid.xml: 1 tests, 0 errors, 1 failures, 0 skipped
```

**Comment by YutoUchimi** [comment]
Travis passed!

**Pull request merged**